client/blocks/importer/hooks/use-site-can-migrate.ts

```diff
@@ -39,11 +42,19 @@ export function useSiteMigrateInfo(
 		}
 	}, [ status, data?.can_migrate ] );
 
+	useEffect( () => {
+		if ( prevIsFetchingData.current === true && isFetchingData === false ) {
+			setIsInitFetchingDone( true );
```

Comment:
👋🏻 Chiming in because we consider state syncing using results from `useQuery` an anti-pattern and an easy source for bugs. I haven't fully reviewed what this code is intended to do but there are other attributes returned from `useQuery` which might be helpful here (e.g. `isInitialLoading` – if you're trying to determine whether this is the first fetch of the query). It'd be awesome to revisit before we ship, I'm more than happy to help and can have a more detailed if desired.

Reply:
@flootr, thanks for the review! I appreciate it.
I'll try with `isInitialLoading`;

Reply:
Since this custom hook has a combination of getting data from the useQuery type of hook and calypso state selector, I don't want to introduce more complexity with more refactoring to this PR (it will be hard to review).

I've created a dedicated ticket to resolve it in a separate PR: https://github.com/Automattic/wp-calypso/issues/84344